File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0] * n
        res[0] = 1
        for i in range(1, n):
            res[i] = res[i - 1] * nums[i - 1]
        temp = 1
        for i in range(n -1, -1, -1):
            res[i] = temp * res[i]
            temp = temp * nums[i]
        return res         

File: Data Structures & Algorithms/climbing-stairs/submission-4.cpp
class Solution {
public:
    int climbStairs(int n) {
        int one = 1;
        int tow = 1;
        for (int i = 0; i < n - 1; i++){
            int temp = tow;
            tow = one;
            one = one + temp;
        }
        return one;
    }
};
